driver-kotlin-sync/build.gradle.kts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import io.gitlab.arturbosch.detekt.Detekt
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.jetbrains.kotlin.jvm") version "1.8.10"
+    `java-library`
+
+    // Test based plugins
+    id("com.diffplug.spotless")
+    id("org.jetbrains.dokka") version "1.7.20"
+    id("io.gitlab.arturbosch.detekt") version "1.21.0"
+}
+
+repositories {
+    mavenCentral()
+    google()
+}
+
+base.archivesName.set("mongodb-driver-kotlin-sync")
+
+description = "The MongoDB Kotlin Driver"
+
+ext.set("pomName", "MongoDB Kotlin Driver")
+
+sourceSets {
+    create("integrationTest") {
+        kotlin.srcDir("$projectDir/src/integration/kotlin")
+        compileClasspath += sourceSets.main.get().output
+        runtimeClasspath += sourceSets.main.get().output
+        compileClasspath += project(":driver-sync").sourceSets.test.get().output
+        runtimeClasspath += project(":driver-sync").sourceSets.test.get().output
+        compileClasspath += project(":driver-core").sourceSets.test.get().output
+        runtimeClasspath += project(":driver-core").sourceSets.test.get().output
+        compileClasspath += project(":bson").sourceSets.test.get().output
+        runtimeClasspath += project(":bson").sourceSets.test.get().output
+    }
+}
+
+val integrationTestImplementation: Configuration by
+    configurations.getting { extendsFrom(configurations.testImplementation.get()) }
+
+dependencies {
+    // Align versions of all Kotlin components
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+
+    api(project(path = ":bson", configuration = "default"))
+    api(project(path = ":driver-sync", configuration = "default"))
+
+    testImplementation("org.jetbrains.kotlin:kotlin-reflect")
+    testImplementation("org.jetbrains.kotlin:kotlin-test-junit")
+    testImplementation("org.mockito.kotlin:mockito-kotlin:4.1.0")
```

Comment:
usually mockk is a better lib for kotlin than mockito-kotlin

Reply:
It does seem nicer but we use Mockito else where, so preferred one mocking library.